scripts/presubmit-tests.sh

```diff
@@ -169,16 +169,22 @@ function default_build_test_runner() {
     # Consider an error message everything that's not a package name.
     errors_go1="$(grep -v '^\(github\.com\|knative\.dev\)/' "${report}" | sort | uniq)"
   fi
-  # Get all build tags in go code (ignore /vendor)
+  # Get all build tags in go test code (ignore /vendor)
   local tags="$(grep -r '// +build' . \
       | grep -v '^./vendor/' | cut -f3 -d' ' | sort | uniq | tr '\n' ' ')"
   if [[ -n "${tags}" ]]; then
     errors=""
-    if ! capture_output "${report}" go test -run=^$ -tags="${tags}" ./... ; then
-      failed=1
-      # Consider an error message everything that's not a successful test result.
-      errors_go2="$(grep -v '^\(ok\|\?\)\s\+\(github\.com\|knative\.dev\)/' "${report}")"
-    fi
+    
+    for pkg in $(go list ./test/...); do
```

Comment:
Nice. But we can go broader, this will build all packages that contain tags (except in vendor). Also a fix: `errors_go2` will accumulate all errors now (not just the last one).

```
    local tagged_pkgs="$(grep -r '// +build' . | grep -v '^./vendor/' | grep ":// +build " | cut -f1 -d: | xargs dirname | sort | uniq | tr '\n' ' ')"
    for pkg in ${tagged_pkgs}; do
      # `go test -c` lets us compile the tests but do not run them.
      if ! capture_output "${report}" go test -c -tags="${tags}" ${pkg} ; then
        failed=1
        # Consider an error message everything that's not a successful test result.
        errors_go2+="$(grep -v '^\(ok\|\?\)\s\+\(github\.com\|knative\.dev\)/' "${report}")"
      fi
      # Remove unused generated binary, if any.
      rm -f e2e.test
    done
```